ext/configuration.h

```diff
@@ -109,6 +109,7 @@ enum ddtrace_dbm_propagation_mode {
     CONFIG(STRING, DD_SPAN_SAMPLING_RULES_FILE, "", .ini_change = ddtrace_alter_sampling_rules_file_config)    \
     CONFIG(SET_LOWERCASE, DD_TRACE_HEADER_TAGS, "")                                                            \
     CONFIG(INT, DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH, "512")                                                     \
+    CONFIG(SET, DD_TRACE_PEER_SERVICE_MAPPING, "")                                                             \
```

Comment:
`peer.service` should be _opt-in_, based on the spec (for now, as we want to enable it by default as we pass private beta). However, this is not a blocker for me to approve this PR, as I can add it the `mysqli` work https://github.com/DataDog/dd-trace-php/pull/2142

Reply:
What should be the default value in this case?

Reply:
`false`

Reply:
Alright, added `DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED`